src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationSource.cs

```diff
@@ -34,13 +36,34 @@ public IConfigurationProvider Build(IConfigurationBuilder builder)
             {
                 AzureAppConfigurationOptions options = _optionsProvider();
 
+                IAzureClientFactory<ConfigurationClient> clientFactory = options.ClientFactory;
+
+                if (options.IsAfdUsed)
+                {
+                    if (options.LoadBalancingEnabled)
```

Comment:
Can you set ReplicaDiscoveryEnabled to false - just to be explicit. I know we cant throw an error because the default is true, but these options are passed on to `AzureAppConfigurationProvider` class. Better to set it to false even if we dont use it later.